### PR TITLE
correct sorting and indexing of level1 data

### PR DESCRIPTION
**Pull request by comane**
This PR responds to https://github.com/NNPDF/nnpdf/issues/1755

See also https://github.com/NNPDF/nnpdf/pull/1757

### CODE REVIEW

**Comment by RoyStegeman** [comment]
To keep some consistency with the implementation of other groupings I believe you should obtain the same index list if you group all datasets into a single metadata group, i.e. grouping by `ALL` instead of `experiment`. Then, maybe instead of  always reverting the ordering you could do it only if the grouping is `ALL` (this is just a suggestion, I haven't really looked into it too much). 

This is labelled a bug, where does this bug manifest itself? Are the covmat or data to which the index is applied not ordered in the same way as the index?

**Comment by Zaharid** [comment]
@RoyStegeman the bug exists in the snippets demostrated by @comane  and myself in #1755. AFAICT the covmats are built group by group and thus not affected but have not looked in detail.

File: validphys2/src/validphys/results.py [hunk 1]
```diff
@@ -151,6 +152,38 @@ def from_convolution(cls, pdf, posset):
         return cls(stats)
 
 
+def data_index(data):
+    """
+    Given a core.DataGroupSpec instance, return pd.MultiIndex
+    with the following levels:
+    
+    1. experiment
```
Comment (Zaharid) [inline validphys2/src/validphys/results.py:160]: This is not correct, as the grouping can be arbitrary.

**Comment by Zaharid** [comment]
The problem with the implementation in this PR is that now the correlations taken into account for level 1 are restricted to be by experiment (thus excluding exotic covariance matrices).

File: validphys2/src/validphys/results.py [hunk 2]
```diff
@@ -151,6 +152,38 @@ def from_convolution(cls, pdf, posset):
...
+    data: core.DataGroupSpec
+
+    Returns
+    -------
+    pd.MultiIndex
+
+    """
+    tuples = []
+
+
```
Comment (scarlehoff) [inline validphys2/src/validphys/results.py:176]:
```suggestion
```

File: validphys2/src/validphys/results.py [hunk 3]
```diff
@@ -151,6 +152,38 @@ def from_convolution(cls, pdf, posset):
...
+    for ds in data.datasets:
+
```
Comment (scarlehoff) [inline validphys2/src/validphys/results.py:178]:
```suggestion
```

**Review by scarlehoff — APPROVED** [review]

**Comment by scarlehoff** [comment]
If you rebase on top of master the tests should pass (I'm tempted to merge even, but maybe it's better to rebase, ensure they pass, then merge)

**Pull request merged**